lib/ecto/repo.ex

```diff
@@ -1055,6 +1055,12 @@ defmodule Ecto.Repo do
   @doc """
   Returns true if the current process is inside a transaction.
 
+  Note that this may not reflect transactions created for tests; since tests
+  almost always run in a transaction, it would almost always report `true`,
+  even when the same code running in production would not be inside a
+  transaction. See the database logs for reliable information on when
+  transactions begin and end.
```

Comment:
I think we can phrase this differently. Because it is not specific to tests but only to the sql sandbox. Something like this:

> If you are using the `Ecto.Adapters.SQL.Sandbox` in tests, note that even though
> each test is inside a transaction, `in_transaction?/1` will only return true if you also
> call `transaction/2`. This is done so the test environment mimics dev and prod.
> However, this may be surprising if you run XYZ...

Thoughts?

Reply:
I was about to comment something similar. Sounds good to me.

Minor nitpick, maybe "will only return true inside transactions explicitly created with `transaction/2`" instead of "will only return true if you also call `transactions/2`?

Reply:
Ah! And should a similar paragraph be added to `Ecto.Adapters.SQL.Sandbox`?